pgmpy/inference/EliminationOrdering.py

```diff
@@ -0,0 +1,101 @@
+class EliminationOrdering(object):
+
+    """
+    This class implements tools to help finding good elimination orderings.
+    Besically, a greedy algorithm scores each variables currently in the
+    moralized graph, pick one, removing it, and score the graph again.
+    The scoring is done by a cost function. From ('Probabilistic Graphical
+    Model Principles and Techniques' - Koller and Friedman), "empirical results
+    show that these heuristic algorithms [min-neighbors, min-fill, min-weight,
+    weighted-min-fill] perform surprisingly well in practice. Generally,
+    Min-Fill and Weighted-Min-Fill tend to work better on more problems".
+
+    Parameters
+    ----------
+    bayesian_model: input bayesian model
+        A bayesian model to be moralized (an undirected graph where the parents
+        of a v-structures are connected).
+    """
+
+    def __init__(self, bayesian_model):
```

Comment:
This class becomes very specific to ```BayesianModel```. Although its rare to see variable elimination applied to ```MarkovModel``` but we want our classes and methods to be as generic as possible. So instead of ```bayesian_model``` use ```model``` and put explicit checks that if it is a ```BayesianModel``` then moralize it.